### PR TITLE
[IA-2226] Add app audit info to JSON response

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/IA-2226

* Adds app `auditInfo` to `GetAppResponse` and `ListAppResponse` and updates Swagger
* Update swagger to include `?deleteDisk=[true|false]` in `deleteApp` docs
* Moved some codecs that are only needed for tests out of `AppRoutes.scala` and into `AppRoutesTestJsonCodec.scala`

Testing on a fiab..

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala [hunk 1]
```diff
@@ -26,6 +28,9 @@ final case class CreateAppRequest(kubernetesRuntimeConfig: Option[KubernetesRunt
                                   labels: LabelMap = Map.empty,
                                   customEnvironmentVariables: Map[String, String])
 
+final case class BatchNodepoolCreateRequest(numNodepools: NumNodepools,
```
Comment (rtitle) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala:31]: Moved this from `leonardoServiceContractModel.scala`, where I think it didn't belong

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/AppRoutes.scala [hunk 2]
```diff
@@ -197,15 +196,6 @@ object AppRoutes {
       } yield CreateAppRequest(c, a.getOrElse(AppType.Galaxy), d, l.getOrElse(Map.empty), cv.getOrElse(Map.empty))
     }
 
-  implicit val nameKeyDecoder: KeyDecoder[ServiceName] = KeyDecoder.decodeKeyString.map(ServiceName.apply)
-  implicit val getAppDecoder: Decoder[GetAppResponse] =
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/AppRoutes.scala:201]: Decoding GET responses and encoding POST requests are really only needed for tests, so I put them in `AppRoutesTestJsonCodec`

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala (hunk 1 above)
Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala:31]: maybe move to https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala?

Comment (rtitle) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala:31]: That's where it is now :)

Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/appRoutesModels.scala:31]: oh..was reading the path wrong

**Review by Qi77Qi — APPROVED** [review]
just curious, what is auditInfo for?

**Comment by rtitle** [comment]
UI team asked for `createdDate` -- but I thought might as well send the whole `auditInfo` structure, same as runtimes and disks (e.g. I think `creator` should be returned too).

**Comment by rtitle** [comment]
Tested in a fiab, both `auditInfo` and `?deleteDisk=true` seem to work

**Pull request merged**